runtime/plugin/net.lua

```diff
@@ -0,0 +1,333 @@
+if not vim.g.lua_net_enable then
+  return
+end
+
+local function url_safe_encode_byte(byte)
+  local char_map = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_'
+  return char_map:sub(byte % 64 + 1, byte % 64 + 1)
+end
+
```

Comment:
These functions should not be defined here (since `plugin/` is executed unconditionally) but in `vim.net`. This file should just define autocommands with callbacks that are (private) functions in `vim.net`.

Reply:
There's also a good chunk of code duplication in these callbacks, which further indicates that these should be refactored into dedicated functions.